src/codeu/chat/client/ClientUser.java

```diff
@@ -65,13 +65,14 @@ public User getCurrent() {
     return current;
   }
 
-  public boolean signInUser(String name) {
+  public boolean signInUser(String name, String password) {
     updateUsers();
 
     final User prev = current;
     if (name != null) {
       final User newCurrent = usersByName.first(name);
-      if (newCurrent != null) {
+
+      if ((newCurrent != null) && (password.equals(newCurrent.password))) {
```

Comment:
Conventionally you wouldn't wrap either side of the boolean in parens, the operator precedence is pretty obvious in this case.